dhis2/dhis2-core
# feat: manually revert a running job to non-running [DHIS2-17419]

### Summary
Adds a new endpoint for `JobConfiguration` to `revert` a job back to a initial state. This is meant to be used after a job gets "stuck" and is considered `RUNNING` but in reality it is not. Then this can be used as a manual way to get that job "unstuck" mostly to clear the way for other jobs of the same type to run.

    POST /api/jobConfigurations/{uid}/revert

The `revert` can be done by the user that executed the job or a user having admin or `F_PERFORM_MAINTENANCE` authority. 

### Automatic Testing
Added scenarios covering basic error paths. The success path is difficult to test as it needs a server stop/start.

### Manual Testing
There is some wired hibernate behaviour where hibernate does an update to the row as a unwanted "side-effect" of `revertNow`. The cause is somehow the loading of the affected row object when checking authority. 

To reproduce:
* goto data administration app
* start the analytics run
* shut down the server while the job runs
* start the server again
* find the ID of the created analytics job
* run `POST /api/jobConfigurations/{uid}/revert` and check the `enabled` flag, it should be `false` but is observed as `true`
* to then have a easier way of recreating the test it is possible to update the row manually 
```sql
update jobconfiguration set jobstatus = 'RUNNING', enabled=true where uid = '{uid}';
```
* don't forget to commit that update, e.g. by reading it back
* now `revert` can be called again as before to check if `enabled=false` stays
